### PR TITLE
Inabox array clear

### PR DESCRIPTION
**Pull request by keithwrightbos**
As part of inabox host script load, verify inbound window objects and clear array after load and ensure it can no longer be populated (free memory).  Added test coverage.

### CODE REVIEW

File: ads/inabox/inabox-host.js [hunk 1]
```diff
@@ -21,30 +21,60 @@
 
 import {dev, initLogConstructor, setReportError} from '../../src/log';
 import {reportError} from '../../src/error';
+import {isArray} from '../../src/types';
 import {InaboxMessagingHost} from './inabox-messaging-host';
 
 const TAG = 'inabox-host';
-run(self);
+const INITIALIZED_WIN_VAR_NAME = 'ampInaboxInitialized';
```
[the remark this answers is not in the document]
Comment (keithwrightbos) [reply ads/inabox/inabox-host.js]: Just trying to be explicit... changed.

File: ads/inabox/inabox-host.js [hunk 2]
```diff
@@ -21,30 +21,60 @@
...
+const PENDING_MESSAGES_WIN_VAR_NAME = 'ampInaboxPendingMessages';
+const INABOX_AD_IFRAMES_IN_VAR_NAME = 'ampInaboxIframes';
```
[the remark this answers is not in the document]
Comment (keithwrightbos) [reply ads/inabox/inabox-host.js]: I assume you mean that INABOX_AD_FRAMES should be INABOX_FRAMES?

File: ads/inabox/inabox-host.js [hunk 3]
```diff
@@ -21,30 +21,60 @@
 
 import {dev, initLogConstructor, setReportError} from '../../src/log';
 import {reportError} from '../../src/error';
+import {isArray} from '../../src/types';
```
[the remark this answers is not in the document]
Comment (keithwrightbos) [reply ads/inabox/inabox-host.js]: It shouldn't matter so long as the compiler is properly minimizing but changed just to be safe

File: ads/inabox/inabox-host.js [hunk 4]
```diff
@@ -23,28 +23,59 @@ import {dev, initLogConstructor, setReportError} from '../../src/log';
...
+            dev().info(TAG, 'Error processing inabox message', message, err);
+          }
+        });
+      } else {
+        dev().info(TAG, `Invalid ${PENDING_MESSAGES}`, queuedMsgs);
+      }
+    }
+    // Empty and ensure that future messages are no longer stored in the array.
+    win[PENDING_MESSAGES] = [];
+    win[PENDING_MESSAGES]['push'] = () => {};
```
Comment (lannka) [inline ads/inabox/inabox-host.js]: is this helpful?

File: ads/inabox/inabox-host.js [hunk 5]
```diff
@@ -23,28 +23,59 @@ import {dev, initLogConstructor, setReportError} from '../../src/log';
...
 
-  win.addEventListener('message', host.processMessage.bind(host));
+    const queuedMsgs = win[PENDING_MESSAGES];
+    if (queuedMsgs) {
+      if (Array.isArray(queuedMsgs)) {
+        queuedMsgs.forEach(message => {
+          try {
+            host.processMessage(message);
+          } catch (err) {
+            dev().info(TAG, 'Error processing inabox message', message, err);
```
Comment (lannka) [inline ads/inabox/inabox-host.js]: probably deserves a dev().error?

File: ads/inabox/inabox-host.js (hunk 4 above)
Comment (keithwrightbos) [reply ads/inabox/inabox-host.js]: Ensures that future calls to push have no effect thereby ensuring memory can be freed.

File: ads/inabox/inabox-host.js (hunk 5 above)
Comment (keithwrightbos) [reply ads/inabox/inabox-host.js]: Done

File: ads/inabox/inabox-host.js (hunk 4 above)
Comment (lannka) [reply ads/inabox/inabox-host.js]: makes sense

**Pull request merged**